docs/user-guide/system-admin/configuration/document-storage-directory.md

```diff
@@ -1,31 +1,27 @@
+Zanata will store files for some uploaded documents on the file system in a configured directory. This directory must be configured by using a system property. Zanata will create the directory if it does not yet exist.
 
-Since files were previously stored as BLOBs in the database, there is a migration operation to move the files from the database to the file system. Migration is performed when liquibase is run, which will happen automatically at server startup, or can be initiated from the command line. If liquibase is run from the command line, the directory must be passed as a parameter to liquibase, in addition to the JNDI name binding.
+Since files were previously stored as BLOBs in the database, there is a migration operation to move the files from the database to the file system. Migration is performed when liquibase is run, which will happen automatically at server startup, or can be initiated from the command line. If liquibase is run from the command line, the directory must be passed as a parameter to liquibase, in addition to the system property.
 
 ## Configuring the Directory (JNDI)
-The path for document storage should be bound to JNDI name ```java:global/zanata/files/document-storage-directory```
+The path for document storage should be set using a system property when starting JBoss.
 
 ### Detailed Procedure
-Open `$JBOSS_HOME/standalone/configuration/standalone.xml` and look for the `naming` subsystem. Add the following section to the `bindings` section:
+Open `$JBOSS_HOME/standalone/configuration/standalone.xml` and look for the `system-properties` section. Add the following property declaration:
 
 ```xml
-<subsystem xmlns="urn:jboss:domain:naming:1.3">
-    ...
-    <bindings>
-        <simple name="java:global/zanata/files/document-storage-directory" value="/example/path"/>
-    </bindings>
-    ...
-</subsystem>
+<property name="zanata.files.documentstorage.directory" value="/example/path"/>
 ```
 
+Alternatively, you can pass this and other system properties to JBoss when starting it (see JBoss documentation for details on how to do this).
+
 ## Running Liquibase from command line
-This section will not explain how to run liquibase from the command line, just how to add the required parameter. The parameter is ```document.storage.directory``` and should be added to the liquibase command in the form ```-Ddocument.storage.directory=/example/path```. The value passed to this parameter should exactly match the value bound to the above JNDI name.
+This section will not explain how to run liquibase from the command line, just how to add the required parameter. The parameter is ```document.storage.directory``` and should be added to the liquibase command in the form ```-Ddocument.storage.directory=/example/path```. The value passed to this parameter should exactly match the value bound to the system property above.
```

Comment:
Well, if we're okay with requiring admins to migrate from the old property
to the new one, we can fail startup if the new property is missing.

https://github.com/zanata/zanata-server/blob/server-3.9.0-rc-1/zanata-war/src/main/resources/db/changelogs/db.changelog-3.1.xml#L265

We can probably work around the checksum problem (if any) using
validCheckSum. The whole setup for that changeset is a bit broken, because
rearranging your filesystem at some point in the future shouldn't cause a
database migration error, long after the documents were migrated from DB to
file system.

Perhaps we should just eliminate the parameter basePathParam and have the
custom code look for the new system property internally (and failing if
it's missing). (This will require the use of validCheckSum.)

So I think we can get by with just the new property.

On 6 May 2016 at 10:13, Carlos Munoz <notifications@github.com> wrote:

> In
> docs/user-guide/system-admin/configuration/document-storage-directory.md
> <https://github.com/zanata/zanata-server/pull/1152#discussion_r62274950>:
>
> >  ## Running Liquibase from command line
> > -This section will not explain how to run liquibase from the command line, just how to add the required parameter. The parameter is ```document.storage.directory``` and should be added to the liquibase command in the form ```-Ddocument.storage.directory=/example/path```. The value passed to this parameter should exactly match the value bound to the above JNDI name.
> > +This section will not explain how to run liquibase from the command line, just how to add the required parameter. The parameter is ```document.storage.directory``` and should be added to the liquibase command in the form ```-Ddocument.storage.directory=/example/path```. The value passed to this parameter should exactly match the value bound to the system property above.
>
> @seanf <https://github.com/seanf> What do you think of supporting the new
> property (so that we have a nice set of zanata-namespaced system
> properties), and internally adding some code to make sure that
> document.storage.directory gets populated with the value of the new
> property. Not sure if that's possible considering the order of execution of
> it all.
>
> Alternatively, and since this will only be a problem when updating Zanata
> from a certain version X to this one, we could make it a documentation
> entry that states that when doing this massive upgrade, you might need to
> start the system with both properties.
>
> Your thoughts?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1152/files/322e50197bb4d13f502a91d47b8e0def0832a7c4#r62274950>
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat